scriptworker/task.py

```diff
@@ -556,12 +556,15 @@ def prepare_to_run_task(context, claim_task):
         int: exit code
 
     """
+    env = deepcopy(os.environ)
+    env['TASK_ID'] = context.task_id or 'None'
```

Comment:
Is it possible to not have a `task_id`?

Reply:
If we haven't claimed a task. It happens in unit tests, not so much in real life.

Reply:
As a nit, should we assume that `context.task_id` is set (`env['TASK_ID'] = context.task_id`), and ensure that unit tests that run this logic are mocked accordingly?
Otherwise, it looks like it could be normal behaviour in real life for a task_id not to be set? :thinking:

Reply:
Could do. Could also move the `or 'None'` logic into the `context.task_id` property. The main issue is that if we don't do this, we get `env['TASK_ID'] = None`, and environment values need to be strings. And we're *currently* getting `"None"` baked into our notarization id strings since we're templatizing it with `os.environ.get("TASK_ID")`.

I wasn't sure how much of a win either of those would be, and I know that people don't love having even more in-test setup and mocking to do before they can run a test, so this or the `context.task_id` returning the string `"None"` seemed like the best amount of improvement while not affecting ease of testing.

Reply:
I'm going to merge this so we get TASK_ID in the notarization id, but if you feel strongly about this, I can address in a followup.

Reply:
> The main issue is that if we don't do this, we get env['TASK_ID'] = None, and environment values need to be strings

I feel like this is pushing an unnecessary extra case down into scripts, since they'll need to understand what it means if `env['TASK_ID']` is set to the magic string "None". If this case should be impossible, perhaps we should validate against it?

> people don't love having even more in-test setup and mocking to do before they can run a test

I agree that test ergonomics are important, but IIUC this solution is weakening the restrictions on state/variables to make tests easier. This will add complexity to the runtime, since there's more potential states to manage.

Reply:
I only know of one scriptworker TASK_ID use case, and that's for use in notarization. In that case, a string of `None` would be no change, and adding a real taskId would be an improvement. If we need to guarantee that TASK_ID is set properly and we foresee issues here, I agree we should test around it.

Noting that this is a case where you're advocating more tests, and I'm pushing back a bit since there are no current real world needs: it's nice to switch roles sometimes :)